ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -574,44 +558,63 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
             tap(address => this.inputAddress$.next(address))
         );
 
-        const emitComponentBusyAction$ = combineLatest([
-            this.isInputComponentBusy$,
-            isThisComponentBusy$
-        ]).pipe(
+        const addressComponentBusy$ = combineLatest([this.isInputComponentBusy$, isThisComponentBusy$]).pipe(
             map(busyFlags => busyFlags.some(val => val)),
             distinctUntilChanged(),
+            share());
+
+        const emitComponentBusyAction$ = addressComponentBusy$.pipe(
+            // let's smooth out our busyness signals
+            debounce(busy => interval(busy ? 0 : 250)),
+            distinctUntilChanged(),
             tap(isBusy => this.componentBusy.emit(isBusy))
         );
 
-        const activitityRequirementsMet$: Observable<boolean> = currentAddress$.pipe(
+        const activityRequirementsMet$: Observable<boolean> = currentAddress$.pipe(
             map(currentAddress => this.meetsActivityRequirements(currentAddress))
         );
 
         const emitValidStatusAction$ = combineLatest([
             this.formValidStatusChanged$,
             this.formErrorMessages$,
             this.verifyFieldErrors$,
-            activitityRequirementsMet$
+            activityRequirementsMet$
         ]).pipe(
             map(([formValidStatusChanged, formErrors, addressErrors, reqsMet]) =>
                 (!this.configuration.addressEnforceRequiredFields || formValidStatusChanged)
                 && !formErrors.length && !addressErrors.length && reqsMet),
             distinctUntilChanged(),
-            tap(status => {
-                this.block.hasValidAddress = status;
-                this.validStatusChanged.emit(status);
-            })
+            tap(status => this.validStatusChanged.emit(status))
         );
 
+        // At least for now, we'll report the status of mailing address through block too
+        // (needed to limit status checks to blocks that are visible within a section
+        const updateBlockValidStatusAction$ = this.validStatusChanged.pipe(
+            tap(validStatus => this.block.hasValidAddress  = validStatus)
+        );
```

Comment:
As the comment indicates, we are setting the same value for validAddress in the block as we are sending via the validStatusChanged.
The code in develop branch was setting the hasValidAddress in different location of code with probably the same value. I am making it explicit here.